private/bufpkg/bufcheck/buflint/internal/buflintbuild/buflintbuild.go

```diff
@@ -348,6 +348,12 @@ var (
 		"all files have a syntax specified",
 		newAdapter(buflintcheck.CheckSyntaxSpecified),
 	)
+	// ValidateRulesTypesMatchRuleBuilder is a rule builder.
+	ValidateRulesTypesMatchRuleBuilder = internal.NewNopRuleBuilder(
+		"VALIDATE_RULES_TYPES_MATCH",
```

Comment:
I don't have a great answer just yet and need to sleep on it, but I don't think this is the best name for this rule - need to brainstorm a bit as to what this rule specifically is and what the name should be. Just a note.

Reply:
modified to `VALIDATE_CONSTRAINTS_CHECK` - all ears for anything better. We also have the option of breaking up the rules for more granular control. I've document all possible rules [here](https://www.notion.so/bufbuild/protovalidate-first-class-buf-lint-support-d96aaa1815694b85b5e6696beded8f55?pvs=4)